evalml/utils/gen_utils.py

```diff
@@ -520,3 +520,64 @@ def deprecate_arg(old_arg, new_arg, old_value, new_value):
         )
         value_to_use = old_value
     return value_to_use
+
+
+def contains_all_ts_parameters(problem_configuration):
+    """Validates that the problem configuration contains all required keys.
+
+    Args:
+        problem_configuration (dict): Problem configuration.
+
+    Returns:
+        bool, str: True if the configuration contains all parameters. If False, msg is a non-empty
+            string with error message.
+    """
+    required_parameters = {"date_index", "gap", "max_delay", "forecast_horizon"}
+    msg = ""
+    if not problem_configuration or not all(
+        p in problem_configuration for p in required_parameters
+    ):
+        msg = (
+            "problem_configuration must be a dict containing values for at least the date_index, gap, max_delay, "
+            f"and forecast_horizon parameters. Received {problem_configuration}."
+        )
+    return not (msg), msg
+
+
+_validation_result = namedtuple(
+    "TSParameterValidationResult",
+    ("is_valid", "msg", "smallest_split_size", "max_window_size"),
+)
+
+
+def are_ts_parameters_valid_for_split(
```

Comment:
I know these are covered in other tests, but can we add some explicit tests for these in the `test_utils` file? Mainly because if we decide to change some stuff in the future, it would be nice to ensure that we have full test coverage for these methods

Reply:
Word adding some small tests now too!